packages/components/src/styles/components/app-frame.scss

```diff
@@ -16,32 +16,47 @@
   grid-template-rows: auto 1fr auto;
   grid-template-columns: auto 1fr;
   min-height: 100vh;
+}
+
+// OPTIONS
 
-  // prevent interaction with AppHeader when SideNav is open/expanded
-  &:has(.hds-side-nav--is-not-minimized) .hds-app-header--is-mobile {
-    pointer-events: none;
+// Make header position sticky/fixed if is-header-fixed is set and the viewport height is greater than 480px
+@media (height >= 480px) {
+  .hds-app-frame--is-header-fixed {
+    .hds-app-frame__header {
+      position: sticky;
+      top: 0;
+      right: 0;
+      left: 0;
+    }
+  
+    // Modify sidenav layout when used together with fixed app-header
+    .hds-app-frame__sidebar {
+      top: var(--token-app-header-height);
+      height: calc(100vh - var(--token-app-header-height));
+      min-height: calc(100vh - var(--token-app-header-height));
+    }
   }
 }
 
-
 // FRAME'S CONTAINERS
 
 .hds-app-frame__header {
-  z-index: 7;
+  z-index: 21; // needs to stay above the main content & sidebar
   grid-area: header;
-
-  // NOTE: May not need to qualify this with .hds-app-header as it may not matter if AppHeader is not present
-  &:has(.hds-app-header) {
-    position: sticky;
-    top: 0;
-    right: 0;
-    left: 0;
-  }
 }
 
 .hds-app-frame__sidebar {
-  z-index: 6;
+  position: sticky;
+  top: 0;
+  z-index: 20; // needs to stay above the main content
   grid-area: sidebar;
+  height: 100vh;
+  min-height: 100vh;
+
+  &:has(.hds-side-nav--is-not-minimized) {
+    z-index: 22; // makes overlay appear above header when sidenav is expanded to block interaction
```

Comment:
This feels weird to me (and "semantically" not correct: the secondary navigation obscures/is above the primary navigation).

Definitely a designers' call, but that's my 2 cents.

Reply:
Also if we remove this, it means we remove the `has(.hds-side-nav--is-not-minimized)` selector which "smells" for me (why should the `AppFrame` know about its children in this way? should be more agnostic, for me)

Reply:
I agree this still needs more input on how this behavior should work. I asked @jorytindall to test and give his feedback. I would also like input from others @hashicorp/hds-design @hashicorp/hds-engineering 

**Problem:**
How should the AppHeader's `MenuButton` appear and behave when the SideNav is expanded in mobile view?

[Test the interactions for yourself](https://hds-showcase-git-hds-3612-app-frame-updates-hashicorp.vercel.app/layouts/app-frame/frameless/demo-full-app-frame-with-app-header)

1. Narrow your browser window until the "Menu" button appears in the top left corner of the AppHeader.
2. Click the SideNav tab to expand the SideNav
3. With the SideNav open, attempt to interact with the "Menu" button.

Is the interaction ok? Confusing? How would you expect the Menu button to behave instead (if differently) and how should it appear visually?

Reply:
I do agree it needs a bit more thought. I asked @jorytindall to test and give his feedback of the behavior. I would appreciate input from other @hashicorp/hds-design people as well.

**To clarify the questions:**
👉 We are trying to figure out the best way of handling the AppHeader's `MenuButton` when the SidNav is expanded (open) in mobile view. How should it appear visually? What should its behavior be?

Reply:
Should we perhaps _not_ disable the `AppHeader` `MenuButton` and `HomeLink` controls at all when the mobile SideNav is expanded?

Reply:
There doesn't appear to be a way to trap focus in both the SideNav & AppHeader simultaneously that I was able to discover. I've instead made a number of refactors to the CSS to include disabled styles for interactive elements in the AppHeader when the SideNav is expanded in mobile view. Using the disabled styles should make it more obvious to users that the MenuButton and HomeLink can't be interacted with.

Reply:
UPDATE: For now, we have decided to add disabled styling to the inoperable controls within the AppHeader to make it clearer to users that they aren't functional.